jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunction.kt

```diff
@@ -69,5 +70,13 @@ data class FunctionUploadDetails(
     val description: String?,
     val envVars: Map<String, String>,
     val timeout: Int,
-    val memorySize: Int
-)
\ No newline at end of file
+    val memorySize: Int,
+    val enableXray: Boolean
+)
+
+fun FunctionUploadDetails.enableXrayAsTracingMode(): TracingMode =
```

Comment:
A function that begins with a verb I would expect a mutation - we're just returning a state it should be a noun.